main.c

```diff
@@ -1269,7 +1269,12 @@ int main(int argc, char **argv) {
 	loop_add_fd(state.eventloop, get_comm_reply_fd(), POLLIN, comm_in, NULL);
 
 	loop_add_fd(state.eventloop, sigusr_fds[0], POLLIN, term_in, NULL);
-	signal(SIGUSR1, do_sigusr);
+
+	struct sigaction sa;
+	sa.sa_handler = do_sigusr;
+	sigemptyset(&sa.sa_mask);
+	sa.sa_flags = SA_RESTART;
+	sigaction(SIGUSR1, &sa, NULL);
```

Comment:
Using `sigaction` is fine, but when I look in glibc and musl, `signal` seems to thin wrappers around `sigaction` (https://git.musl-libc.org/cgit/musl/tree/src/signal/signal.c, https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/signal.c;h=8b3681558b7e637397e62945bef58770d08e20bb;hb=HEAD). Why would it behave differently?